src/jspdf.js

```diff
@@ -4850,14 +4850,50 @@ function jsPDF(options) {
    * @param {string} postScriptName PDF specification full name for the font.
    * @param {string} id PDF-document-instance-specific label assinged to the font.
    * @param {string} fontStyle Style of the Font.
+   * @param {number || string} fontWeight Weight of the Font.
    * @param {Object} encoding Encoding_name-to-Font_metrics_object mapping.
    * @function
    * @instance
    * @memberof jsPDF#
    * @name addFont
    * @returns {string} fontId
    */
-  API.addFont = function(postScriptName, fontName, fontStyle, encoding) {
+  API.addFont = function(
+    postScriptName,
+    fontName,
+    fontStyle,
+    fontWeight,
+    encoding
+  ) {
+    let encodingOptions = [
+      "StandardEncoding",
+      "MacRomanEncoding",
+      "Identity-H",
+      "WinAnsiEncoding"
+    ];
+    if (arguments[3] && encodingOptions.indexOf(arguments[3]) !== -1) {
+      //IE 11 fix
+      encoding = arguments[3];
+    } else if (arguments[3] && encodingOptions.indexOf(arguments[3]) == -1) {
+      //if weired combination of fontweight and font style throw error
+      if (
+        (fontStyle == "normal" && fontWeight == "bold") ||
+        (fontStyle == "bold" && fontWeight == "normal") ||
+        (fontStyle == "bold" && fontWeight == 400) ||
+        (fontStyle == "normal" && fontWeight == 700)
+      ) {
+        throw new Error("Invalid Combination of fontweight and fontstyle");
+      }
+      if (fontWeight && fontStyle !== fontWeight) {
+        //if fontstyle is normal and fontweight is normal too no need to append the font-weight
+        fontStyle =
+          fontWeight == 400
+            ? "normal"
+            : fontWeight == 700
+            ? "bold"
+            : fontStyle + "" + fontWeight;
```

Comment:
What about "italic" fontStyle? E.g. "italic" + "400" should result in "italic".

Reply:
Done.